### PR TITLE
Merge MRO

### PR DESCRIPTION
**Pull request by michaelsauter**
Merges https://github.com/opendevstack/ods-mro-jenkins-shared-library into this repo, preserving the history of the MRO repo.

Closes https://github.com/opendevstack/ods-mro-jenkins-shared-library/issues/104.

As the diff is impossible to review, here's a summary of what I've done:

- Merge commits (d90641c). There were only a few conflicts. I solved the conflict on the `README.md` by creating a new document under `docs` for the orchestration pipeline.
- Rename `mroPipeline` to `odsOrchestrationPipeline` (7037ad8)
- Move all packages from MRO in package `org.ods` into `org.ods.orchestration` (in 596928b, tests moved in c89e93c)
- Move all "phases" in `vars/` to stages in `org.ods.orchestration` (7e30ebe)
- Make use of the super stage to contain the try/catch logic with reporting back to Jira (dadf8df)
- Move `runOnAgentPod` into super stage (660a27b)
- Fix issues that occurred due to the merge (5eb4238, and a regression on MRO `master` in 
313787d)

Far from being complete, but this allows to use the orchestration pipeline like this:
```groovy
@Library('ods-jenkins-shared-library@master') _
odsOrchestrationPipeline(
  debug: true,
  odsImageTag: 'master'
)
```

I've recorded a few follow-up items: #272, #273, #274, #275 and #276 which should cover most things related to the merge of the two libraries.

I propose to merge this quickly to unblock contributions to the MRO again.

### CODE REVIEW

File: src/org/ods/orchestration/usecase/LeVADocumentUseCase.groovy [hunk 1]
```diff
@@ -0,0 +1,1380 @@
...
+            ],
+            data         : [
+                repo    : repo,
+                sections: sections
+            ]
+        ]
+
+        // Code review report - in the special case of NO jira ..
+        def codeReviewReport
+        if (!this.jiraUseCase.jira &&
```
Comment (clemensutschig) [inline src/org/ods/orchestration/usecase/LeVADocumentUseCase.groovy:1134]: misses (this.project.isAssembleModel())

Comment (michaelsauter) [reply src/org/ods/orchestration/usecase/LeVADocumentUseCase.groovy]: Fixed in 313787d.

**Comment by metmajer** [comment]
Fantastic @michaelsauter! Did you run a release manager pipeline with the new codebase already? Ping me on IM if you like.

**Comment by michaelsauter** [comment]
@metmajer Yes. Assembling (WIP and version) worked, and promotion as well. I did not test the Jira integration, but since I did not any changes there things should work :)

**Pull request merged**